roles/roles_common.js

```diff
@@ -306,9 +315,9 @@ _.extend(Roles, {
    */
```

Comment:
Please take a look at this function and make sure there is no regression happening. Tests all pass, but this is different enough from the underscore version that it's worth getting a few more eye balls on it.